### PR TITLE
simple email validity check for class registration

### PR DESCRIPTION
**Pull request by Fury-101**
fixes https://github.com/cpinitiative/cpinitiative/issues/103

### CODE REVIEW

**Comment by maggie-j-liu** [comment]
did you test this?

**Comment by Fury-101** [comment]
having issues building the project but i've tested that the type="email" works to catch misspellings (ie ,com)

**Comment by maggie-j-liu** [comment]
hmm, what issues are you having w/ building the project? (also try testing on the deployed version ^)

**Comment by Fury-101** [comment]
okay it seems to not work since the inputs arent in a form

**Comment by Fury-101** [comment]
this should now work

**Comment by ADawesomeguy** [comment]
> having issues building the project but i've tested that the type="email" works to catch misspellings (ie ,com)

Is this not browser dependent? As in left up to the client-sided implementation? I think this should be fine but maybe in the future it would be better to implement some sort of regex pattern to check instead.

**Comment by Fury-101** [comment]
It is indeed left up to the browser but most browsers have a regex check - i dont think it’s different across browsers

**Comment by ADawesomeguy** [comment]
Yep, seems like it is client sided but follows some sort of standard where it is implemented. Should definitely be good enough for a preliminary check 👍🏽.

File: src/components/classes/registration/StudentInformationSection.tsx [hunk 1]
```diff
@@ -68,9 +68,9 @@ export default function StudentInformationSection({
                     Student Email Address
                   </label>
                   <input
-                    type="text"
+                    type="email"
                     value={email}
-                    onChange={e => setEmail(e.target.value)}
+                    onChange={e => {setEmail(e.target.value); this.reportValidity()}}
```
Comment (maggie-j-liu) [inline src/components/classes/registration/StudentInformationSection.tsx:73]: uhh just tested; doesn't seem to work...

what is `this` ?

**Comment by maggie-j-liu** [comment]
@Fury-101 what issues are you having when building the project? i can help -- you should be running locally so you can test it

File: src/components/classes/registration/StudentInformationSection.tsx (hunk 1 above)
Comment (Fury-101) [reply src/components/classes/registration/StudentInformationSection.tsx:73]: How should i be installing yhe dependencies? Npm throws an error when installing

Comment (maggie-j-liu) [reply src/components/classes/registration/StudentInformationSection.tsx:73]: install with `yarn` (if you don't have yarn, `npm install -g yarn` should work)

Comment (ADawesomeguy) [reply src/components/classes/registration/StudentInformationSection.tsx:73]: I think Yarn docs recommend using corepack (either installing it with `npm i -g corepack` or enabling it with `corepack enable`.

**Comment by Fury-101** [comment]
alright so i've built it and this does work, however it's spammier than I thought (notification pops up every time you type a character).
the original intent was to listen to the onChange event but after further research react changes onchange to oninput, so I think that's the best possible solution

**Comment by maggie-j-liu** [comment]
hmmm i agree that it's a bit spammy... i think what you could do is listen to the onBlur event? wouldn't catch all cases but we can check once again when submitting & report any issues then

and also, it doesn't stop the form from being submitted with an invalid email; could you add some error message when submitting if the email isn't valid, maybe right below the submit button? like the payment thing does this, but add it for the financial aid submission button too? 
image

**Comment by Fury-101** [comment]
hmm i thought i was listening to onBlur before and it seemed to have the same behavior
now it uses onblur though

**Comment by Fury-101** [comment]
also, ideally we would also add an error message when submitting if the email isn't valid but onblur prevents the element from losing focus if the email is invalid so they _have_ to fix it

**Comment by Fury-101** [comment]
this should now work

**Comment by SansPapyrus683** [comment]
so that was a lie

**Comment by thecodingwizard** [comment]
Oops I really should have merged this earlier.. Thanks!

**Pull request merged**